Fix `planner_llm_model_name` Not Loaded Correctly When Using "Load Config"

knowlet opened the pull request:
## Solution
- In the WebuiManager.load_config method, retrieve the value of `planner_llm_provider` before setting the model name.
- When setting the value for `planner_llm_model_name`, dynamically update its choices based on the provider, and use `gr.Dropdown(value=..., choices=..., allow_custom_value=True)` to update the component.
- All other components remain unchanged.

## Impact
- When using the "Load Config" feature, the planner_llm_model_name dropdown will now correctly display the model name according to the loaded configuration, without requiring manual reselection.
- This approach can be extended to other dropdowns with similar dependencies if needed.

## Testing
1. In the UI, select different values for planner_llm_provider and planner_llm_model_name, then save the configuration.
2. Restart the WebUI or switch configuration files, and use "Load Config" to load the saved configuration.
3. Verify that planner_llm_model_name correctly displays the corresponding model name, and that the provider and model name are properly synchronized.

## Screenshot
<img width="1009" alt="image" src="https://github.com/user-attachments/assets/f76541ef-7f68-41cc-8c37-e72f46bc4df1" />

## Reference
fix #589
    
<!-- This is an auto-generated description by mrge. -->
---

## Summary by mrge
Fixed the "Load Config" feature so the planner LLM model name dropdown now shows the correct model based on the selected provider.

- **Bug Fixes**
  - The dropdown updates its choices and value automatically when loading a config, matching the saved provider and model name.

<!-- End of auto-generated description by mrge. -->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/browser-use/web-ui?pullRequest=590) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/browser-use/web-ui?pullRequest=590) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/browser-use/web-ui?pullRequest=590) it.</sub>

vvincent1234 commented:
planner_llm_model_name  should set to None because we don't want llm_panner in default

knowlet commented:
Change the default back to "None" and fix it by yielding an update first.
![image](https://github.com/user-attachments/assets/27bc2eaf-5d3d-42b4-bfdf-f769c41738c2)
![image](https://github.com/user-attachments/assets/baecdf6e-3bbd-4e28-84f0-357abf24b638)

vvincent1234 commented:
Thank you. You don't need a async , just use import time and time.sleep(0.2), i will let this pr pass

knowlet commented:
You're right, I've changed back to not using async and using time.Sleep.

time.Sleep didn't work the first time I tested it because I forgot to yield update_components. I switched to async while testing other methods and forgot to switch back. Thanks for the reminder!
